packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection, Org, SfProject } from '@salesforce/core';
+import {
+  SourceTracking,
+  SourceTrackingOptions
+} from '@salesforce/source-tracking';
+
+export class SourceTrackingService {
+  public static async createSourceTracking(
+    projectPath: string,
+    connection: Connection
+  ): Promise<SourceTracking> {
+    // Since SourceTracking is initialized with an SfProject, which
+    // contains the project path, and PR #4643 made it so that VSCE is
+    // running with process.cwd set as the project root, there
+    // is no need to call process.chdir here as has been done in VSCE
+    // with other core types like Config and ConfigAggregator.
+    const project = await SfProject.resolve(projectPath);
+    const org = await Org.create({ connection });
+    const options: SourceTrackingOptions = {
+      org,
+      project,
+      ignoreLocalCache: true,
```

Comment:
The [docs in the stl library](https://github.com/forcedotcom/source-tracking/blob/5c29282008987a778a83bc27e6030d0ffad28990/src/sourceTracking.ts#L68) say that this should be set to false when used in long-running processes, however, when I set this to false there is the the issue where it doesn't recognize new files and allows conflicts to be overridden.  To get it to recognize newly created files, I had to set `ignoreLocalCache` to true here and make the call to `ensureLocalTracking()` later after creating the instance.

Reply:
That comment definitely seems like it was made for us 😄 do we need to file a bug there slash is that already done?